tests/test_backends.py

```diff
@@ -17,22 +17,30 @@
 from instructlab.model.backends.vllm import build_vllm_cmd
 
 
-# helper function to create dummy valid and invalid safetensor model directories
-def create_safetensors_model_files(safetensors_model_path: pathlib.Path, valid: bool):
+# helper function to create dummy valid and invalid safetensor or bin model directories
+def create_safetensors_or_bin_model_files(
+    model_path: pathlib.Path, model_file_type: str, valid: bool
+):
     test_json_dict = {"a": 1, "b": 2}
     json_object = json.dumps(test_json_dict, indent=4)
 
     for file in ["tokenizer.json", "tokenizer_config.json"]:
-        os.makedirs(os.path.dirname(safetensors_model_path / file), exist_ok=True)
-        with open(safetensors_model_path / file, "a+", encoding="UTF-8") as f:
+        os.makedirs(os.path.dirname(model_path / file), exist_ok=True)
+        with open(model_path / file, "a+", encoding="UTF-8") as f:
             f.write(json_object)
 
-    with open(
-        safetensors_model_path / "model.safetensors", "a+", encoding="UTF-8"
-    ) as f:
+    model_file = (
```

Comment:
```
with open(model_path / f"model.{model_file_type}", ...):
    pass
```

? (`pass` will be equivalent to `f.write("")` I think.)

Reply:
what would the benefit of making that change be?

Reply:
2 lines vs 10 lines? The nested ternary `if-else` is also rather convoluted.

Reply:
Personally I'm not a fan of this approach, as it increases the valid values for file types from just `safetensors` and `bin` to allow all valid strings.

We should instead keep the values constricted to just `safetensors` and `bin` as those are types which we explicitly support, and instead throw a `ValueError` if the provided model type ever deviates.

In all likelihood, this function shouldn't be used outside of this codebase or even that much, so we shouldn't run into this error in the first place.

Reply:
So we can just make the following change:
```suggestion
    if model_file_type not in ("safetensors", "bin"):
        raise ValueError(f"model file type is not supported: {model_file_type}")
    with open(os.path.join(model_path, f"model.{model_file_type}"), "a+", encoding="UTF-8"):
        pass
```

Reply:
since this is test case, and if you'd like to raise an error, you can also `assert model_file_type in (...)`

Reply:
(that said, I think there's a test variant here that actually passes "other".)

Reply:
@booxter ah sorry - I totally misread what you were suggesting 
have updated the code to match that, thanks! 🙏 

@RobotSail this is just a test case so I think we do need to create valid and invalid scenarios to properly test the code